application-load-balancer/lambda_function/secret_rotator.py

```diff
@@ -0,0 +1,625 @@
+                aliases = distribution.get("Aliases", {}).get("Items", [])
+                if any(domain in aliases for domain in self.distro_list.split(",")):
+                    matching_distributions.append({
+                        "Id": distribution["Id"],
+                        "Origin": distribution['Origins']['Items'][0]['DomainName'],
+                        "Domain": distribution['Aliases']['Items'][0]
+                    })
+        logger.info(f"Matched cloudfront distributions: {matching_distributions}")
+        return matching_distributions
+
+    def get_waf_acl(self) -> Dict[str, Any]:
+        client = boto3.client('wafv2')
+        return client.get_web_acl(Name=self.waf_acl_name, Scope='REGIONAL', Id=self.waf_acl_id)
+
+    def _create_byte_match_statement(self, search_string: str) -> Dict[str, Any]:
+        return {
+            'ByteMatchStatement': {
+                'FieldToMatch': {'SingleHeader': {'Name': self.header_name}},
+                'PositionalConstraint': 'EXACTLY',
+                'SearchString': search_string,
+                'TextTransformations': [{'Type': 'NONE', 'Priority': 0}]
+            }
+        }
+
+    def update_waf_acl(self, new_secret: str, prev_secret: str) -> None:
+        client = boto3.client('wafv2')
+        waf_acl = self.get_waf_acl()
+        lock_token = waf_acl['LockToken']
+        metric_name = f"{self.application}-{self.environment}-XOriginVerify"
+        rule = {
+            'Name': f"{self.application}{self.environment}XOriginVerify",
+            'Priority': self.waf_rule_priority,
+            'Action': {'Allow': {}},
+            'VisibilityConfig': {
+                'SampledRequestsEnabled': True,
+                'CloudWatchMetricsEnabled': True,
+                'MetricName': metric_name
+            },
+            'Statement': {
+                'OrStatement': {
+                    'Statements': [
+                        self._create_byte_match_statement(new_secret),
+                        self._create_byte_match_statement(prev_secret)
+                    ]
+                }
+            }
+        }
+
+        new_rules = [rule] + [r for r in waf_acl['WebACL']['Rules'] if r['Priority'] != self.waf_rule_priority]
+        logger.info("Updating WAF WebACL with new rules.")
+
+        response = client.update_web_acl(
+            Name=self.waf_acl_name,
+            Scope='REGIONAL',
+            Id=self.waf_acl_id,
+            LockToken=lock_token,
+            DefaultAction={'Block': {}},
+            Description='CloudFront Origin Verify',
+            VisibilityConfig={
+                'SampledRequestsEnabled': True,
+                'CloudWatchMetricsEnabled': True,
+                'MetricName': metric_name
+            },
+            Rules=new_rules
+        )
+
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+            logger.info("WAF WebACL rules updated")
+
+    def get_cf_distro(self, distro_id: str) -> Dict:
```

Comment:
If the client was passed into the SecretRotator class, then this function and the one below could be removed as they are just wrappers for the client call

Reply:
This can be picked up in a later PR.  We want to maintain the structure for now so we dont break our tests.